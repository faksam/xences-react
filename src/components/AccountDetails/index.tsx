// import libraries
import * as React from 'react';

// third-party libraries

// styles

// components
import { Button, Col, Row, TextInput } from 'react-materialize';

// interfaces

// helper function

const AccountDetails = () => {
  return (
    <React.Fragment>
      <Row>
        <h4>Account Details</h4>
        <TextInput className="formInput" label="Display Name" />
        <TextInput className="formInput" label="Phone No" />
        <TextInput className="formInput" label="Address" />
        <TextInput className="formInput" email validate label="Email" />
      </Row>
      <Row>
        <Col s={4}><Button className="center">Update</Button></Col>
        <Col s={4}><Button className="center">Change Password</Button></Col>
      </Row>
    </React.Fragment>
  );
};

export default AccountDetails;
