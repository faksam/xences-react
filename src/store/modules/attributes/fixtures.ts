export const attributes = [
  {
    id: '1',
    name: 'color',
    description: 'Product color',
    createdAt:  '12/05/19',
  },
  {
    id: '2',
    name: 'size',
    description: 'Product size',
    createdAt:  '12/05/19',
  },
  {
    id: '3',
    name: 'color',
    description: 'Product color',
    createdAt:  '12/05/19',
  },
];
