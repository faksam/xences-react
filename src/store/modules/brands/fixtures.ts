export const brands = [
  {
    id: '1',
    name: 'Nike',
    description: 'Brand Nike',
    createdAt:  '12/05/19',
  },
  {
    id: '2',
    name: 'Puma',
    description: 'Brand Puma',
    createdAt:  '12/05/19',
  },
  {
    id: '3',
    name: 'Otunba',
    description: 'Brand Otunba',
    createdAt:  '12/05/19',
  },
];

export const colors = [
  {
    id: '1',
    name: 'Black',
    description: 'Black',
    createdAt:  '12/05/19',
  },
  {
    id: '2',
    name: 'Red',
    description: 'Red',
    createdAt:  '12/05/19',
  },
  {
    id: '3',
    name: 'Pink',
    description: 'Pink',
    createdAt:  '12/05/19',
  },
  {
    id: '4',
    name: 'White',
    description: 'White',
    createdAt:  '12/05/19',
  },
  {
    id: '5',
    name: 'Blue',
    description: 'Blue',
    createdAt:  '12/05/19',
  },
];
