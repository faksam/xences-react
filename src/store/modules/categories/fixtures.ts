export const categories = [
  {
    id: '1',
    name: 'Cat 1',
    description: 'Category One',
    createdAt:  '12/05/19',
  },
  {
    id: '2',
    name: 'Cat 2',
    description: 'Category Two',
    createdAt:  '12/05/19',
  },
  {
    id: '3',
    name: 'Cat 3',
    description: 'Category Three',
    createdAt:  '12/05/19',
  },
];
