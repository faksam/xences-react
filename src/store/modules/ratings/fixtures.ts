export const ratings = [
  {
    id: '1',
    rating: '4',
    createdAt:  '12/05/19',
  },
  {
    id: '2',
    rating: '4',
    createdAt:  '12/05/19',
  },
  {
    id: '3',
    rating: '4',
    createdAt:  '12/05/19',
  },
];
