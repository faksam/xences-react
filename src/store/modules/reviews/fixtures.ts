export const reviews = [
  {
    id: '1',
    review: 'Good product',
    createdAt:  '12/05/19',
  },
  {
    id: '2',
    review: 'Good product',
    createdAt:  '12/05/19',
  },
  {
    id: '3',
    review: 'Good product',
    createdAt:  '12/05/19',
  },
];
