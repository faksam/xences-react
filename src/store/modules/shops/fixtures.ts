// fixtures
import { users } from '../../modules/users/fixtures';

export const shops = [
  {
    id: '1',
    name: 'Shop 1',
    description: 'Shop One',
    address: 'Cosmos',
    owner: users[0],
    createdAt:  '12/05/19',
  },
  {
    id: '2',
    name: 'Shop 2',
    description: 'Shop Two',
    address: 'Cosmos',
    owner: users[0],
    createdAt:  '12/05/19',
  },
  {
    id: '3',
    name: 'Shop 3',
    description: 'Shop Three',
    address: 'Cosmos',
    owner: users[0],
    createdAt:  '12/05/19',
  },
];
