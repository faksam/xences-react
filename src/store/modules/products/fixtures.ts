// fixtures
import { attributes } from '../../modules/attributes/fixtures';
import { brands } from '../../modules/brands/fixtures';
import { categories } from '../../modules/categories/fixtures';
import { ratings } from '../../modules/ratings/fixtures';
import { reviews } from '../../modules/reviews/fixtures';
import { shops } from '../../modules/shops/fixtures';

export const products = [
  {
    attributes,
    reviews,
    ratings,
    id: '1',
    name: 'Nice shirt',
    discountPercent: '10',
    price: '2000',
    gender: 'male',
    brand: brands[0],
    quantity: 7,
    pictures: 'https://res.cloudinary.com/faksam-soft/image/upload/v1559209413/Man_005_shaco9.jpg',
    shop: shops[0],
    category: categories[1].name,
    createdAt: '05/12/19',
  },
  {
    attributes,
    reviews,
    ratings,
    id: '2',
    name: 'Nice cap',
    discountPercent: '10',
    price: '2000',
    gender: 'male',
    brand: brands[0],
    quantity: 7,
    pictures: 'https://res.cloudinary.com/faksam-soft/image/upload/v1559209413/Man_005_shaco9.jpg',
    shop: shops[0],
    category: categories[1].name,
    createdAt: '05/12/19',
  },
  {
    attributes,
    reviews,
    ratings,
    id: '3',
    name: 'Nice coat',
    discountPercent: '10',
    price: '2000',
    gender: 'male',
    brand: brands[0],
    quantity: 7,
    pictures: 'https://res.cloudinary.com/faksam-soft/image/upload/v1559209413/Man_005_shaco9.jpg',
    shop: shops[0],
    category: categories[1].name,
    createdAt: '05/12/19',
  },
  {
    attributes,
    reviews,
    ratings,
    id: '4',
    name: 'Nice top',
    discountPercent: '10',
    price: '2000',
    gender: 'male',
    brand: brands[0],
    quantity: 7,
    pictures: 'https://res.cloudinary.com/faksam-soft/image/upload/v1559209413/Man_005_shaco9.jpg',
    shop: shops[0],
    category: categories[1].name,
    createdAt: '05/12/19',
  },
  {
    attributes,
    reviews,
    ratings,
    id: '5',
    name: 'Nice skirt',
    discountPercent: '10',
    price: '2000',
    gender: 'male',
    brand: brands[0],
    quantity: 7,
    pictures: 'https://res.cloudinary.com/faksam-soft/image/upload/v1559209413/Man_005_shaco9.jpg',
    shop: shops[0],
    category: categories[1].name,
    createdAt: '05/12/19',
  },
  {
    attributes,
    reviews,
    ratings,
    id: '6',
    name: 'Nice shirt',
    discountPercent: '10',
    price: '2000',
    gender: 'male',
    brand: brands[0],
    quantity: 7,
    pictures: 'https://res.cloudinary.com/faksam-soft/image/upload/v1559209413/Man_001_xaepru.jpg',
    shop: shops[0],
    category: categories[1].name,
    createdAt: '05/12/19',
  },
  {
    attributes,
    reviews,
    ratings,
    id: '7',
    name: 'Nice cap',
    discountPercent: '10',
    price: '2000',
    gender: 'male',
    brand: brands[0],
    quantity: 7,
    pictures: 'https://res.cloudinary.com/faksam-soft/image/upload/v1559209413/Man_001_xaepru.jpg',
    shop: shops[0],
    category: categories[1].name,
    createdAt: '05/12/19',
  },
  {
    attributes,
    reviews,
    ratings,
    id: '8',
    name: 'Nice coat',
    discountPercent: '10',
    price: '2000',
    gender: 'male',
    brand: brands[0],
    quantity: 7,
    pictures: 'https://res.cloudinary.com/faksam-soft/image/upload/v1559209413/Man_001_xaepru.jpg',
    shop: shops[0],
    category: categories[1].name,
    createdAt: '05/12/19',
  },
  {
    attributes,
    reviews,
    ratings,
    id: '9',
    name: 'Nice top',
    discountPercent: '10',
    price: '2000',
    gender: 'male',
    brand: brands[0],
    quantity: 7,
    pictures: 'https://res.cloudinary.com/faksam-soft/image/upload/v1559209413/Man_001_xaepru.jpg',
    shop: shops[0],
    category: categories[1].name,
    createdAt: '05/12/19',
  },
  {
    attributes,
    reviews,
    ratings,
    id: '10',
    name: 'Nice skirt',
    discountPercent: '10',
    price: '2000',
    gender: 'male',
    brand: brands[0],
    quantity: 7,
    pictures: 'https://res.cloudinary.com/faksam-soft/image/upload/v1559209413/Man_001_xaepru.jpg',
    shop: shops[0],
    category: categories[1].name,
    createdAt: '05/12/19',
  },
];

export const menClothings = [
  {
    id: '1',
    name: 'Shirts',
    description: 'Shirts',
    createdAt:  '12/05/19',
  },
  {
    id: '2',
    name: 'T-Shirts',
    description: 'T-Shirts',
    createdAt:  '12/05/19',
  },
  {
    id: '3',
    name: 'Polo Shirts',
    description: 'Polo Shirts',
    createdAt:  '12/05/19',
  },
  {
    id: '4',
    name: 'Pants',
    description: 'Pants',
    createdAt:  '12/05/19',
  },
  {
    id: '5',
    name: 'Jeans',
    description: 'Jeans',
    createdAt:  '12/05/19',
  },
  {
    id: '6',
    name: 'Jumpers - Cardigans',
    description: 'Jumpers - Cardigans',
    createdAt:  '12/05/19',
  },
  {
    id: '7',
    name: 'Hoodies - Sweatshirts',
    description: 'Hoodies - Sweatshirts',
    createdAt:  '12/05/19',
  },
  {
    id: '8',
    name: 'Blazers - Suits',
    description: 'Blazers - Suits',
    createdAt:  '12/05/19',
  },
  {
    id: '9',
    name: 'Jackets - Coats',
    description: 'Jackets - Coats',
    createdAt:  '12/05/19',
  },
  {
    id: '10',
    name: 'Underwear - Sleepwear',
    description: 'Underwear - Sleepwear',
    createdAt:  '12/05/19',
  },
  {
    id: '10',
    name: 'Beachwear',
    description: 'Beachwear',
    createdAt:  '12/05/19',
  },
];
