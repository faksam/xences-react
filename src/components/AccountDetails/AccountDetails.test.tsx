// react libraries
import * as React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';

// third-party libraries
import { mount } from 'enzyme';

// components
import AccountDetails from './';

// fixtures
import { products } from '../../store/modules/products/fixtures';

describe('The AccountDetails component', () => {
  describe('Successful displays', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = mount(
        <Router>
          <AccountDetails />
        </Router>
      );
    });

    it('should be render products page', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
