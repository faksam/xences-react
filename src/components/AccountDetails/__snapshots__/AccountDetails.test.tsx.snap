// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The AccountDetails component Successful displays should be render products page 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <AccountDetails>
      <Row>
        <div
          className="row"
        >
          <h4>
            Account Details
          </h4>
          <TextInput
            className="formInput"
            label="Display Name"
          >
            <div
              className="input-field col"
            >
              <input
                className="formInput"
                id={2}
                type="text"
              />
              <label
                className=""
                htmlFor={2}
              >
                Display Name
              </label>
              <span
                className="helper-text"
              />
            </div>
          </TextInput>
          <TextInput
            className="formInput"
            label="Phone No"
          >
            <div
              className="input-field col"
            >
              <input
                className="formInput"
                id={3}
                type="text"
              />
              <label
                className=""
                htmlFor={3}
              >
                Phone No
              </label>
              <span
                className="helper-text"
              />
            </div>
          </TextInput>
          <TextInput
            className="formInput"
            label="Address"
          >
            <div
              className="input-field col"
            >
              <input
                className="formInput"
                id={4}
                type="text"
              />
              <label
                className=""
                htmlFor={4}
              >
                Address
              </label>
              <span
                className="helper-text"
              />
            </div>
          </TextInput>
          <TextInput
            className="formInput"
            email={true}
            label="Email"
            validate={true}
          >
            <div
              className="input-field col"
            >
              <input
                className="formInput"
                id="email6"
                type="email"
              />
              <label
                className=""
                htmlFor="email6"
              >
                Email
              </label>
              <span
                className="helper-text"
              />
            </div>
          </TextInput>
        </div>
      </Row>
      <Row>
        <div
          className="row"
        >
          <Col
            s={4}
          >
            <div
              className="col s4"
            >
              <Button
                className="center"
                node="button"
              >
                <button
                  className="btn center"
                  disabled={false}
                >
                  Update
                </button>
              </Button>
            </div>
          </Col>
          <Col
            s={4}
          >
            <div
              className="col s4"
            >
              <Button
                className="center"
                node="button"
              >
                <button
                  className="btn center"
                  disabled={false}
                >
                  Change Password
                </button>
              </Button>
            </div>
          </Col>
        </div>
      </Row>
    </AccountDetails>
  </Router>
</BrowserRouter>
`;
